### PR TITLE
forgerock: use dynamic mappings for object fields

### LINKED ISSUE #8046: [forgerock] Use more specific field mappings
There are several instances of fields mapped with `type: object` in the ForgeRock integration. A future version of package-spec will disallow these generic mappings. To prevent conflicts, we want to prefer a mappings that specify the field types (these can be dynamic mappings) or a mapping where the type does not matter (flattened).

```
# 'type: object' mappings
packages/forgerock/data_stream/am_access/fields/forgerock-fields.yml:16:3 forgerock.request.detail
packages/forgerock/data_stream/am_access/fields/forgerock-fields.yml:43:3 forgerock.response.detail
packages/forgerock/data_stream/am_access/fields/forgerock-fields.yml:76:3 forgerock.http.request.queryParameters
packages/forgerock/data_stream/am_access/fields/forgerock-fields.yml:79:3 forgerock.http.request.headers
packages/forgerock/data_stream/am_activity/fields/forgerock-fields.yml:7:3 forgerock.before
packages/forgerock/data_stream/am_activity/fields/forgerock-fields.yml:10:3 forgerock.before.sunAMAuthInvalidAttemptsData
packages/forgerock/data_stream/am_activity/fields/forgerock-fields.yml:13:3 forgerock.after
```

Relates: https://github.com/elastic/package-spec/pull/628

### PR DESCRIPTION
**Pull request by efd6**
## Proposed commit message

See title.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

- Closes #8046

### CODE REVIEW

File: packages/forgerock/data_stream/am_access/fields/forgerock-fields.yml [hunk 1]
```diff
@@ -13,8 +13,10 @@
 - name: forgerock.realm
   type: keyword
   description: The realm where the operation occurred.
-- name: forgerock.request.detail
+- name: forgerock.request.detail.*
   type: object
+  object_type: keyword
+  object_type_mapping_type: '*'
   description: Details around the response status.
 - name: forgerock.request.detail.action
```
Comment (efd6) [inline packages/forgerock/data_stream/am_access/fields/forgerock-fields.yml:21]: I left the `forgerock.request.detail.*` fields here and below for the documentation they provide. It's not clear to me whether this breaks the dynamic mapping. The same situation is in the `forgerock.response.detail.*` fields below.
```
      "dynamic_templates": [
        {
          "forgerock.http.request.queryParameters": {
            "path_match": "forgerock.http.request.queryParameters.*",
            "mapping": {
              "type": "keyword"
            }
          }
        },
        {
          "strings_as_keyword": {
            "match_mapping_type": "string",
            "mapping": {
              "ignore_above": 1024,
              "type": "keyword"
            }
          }
        }
      ],
```

File: packages/forgerock/data_stream/am_activity/fields/forgerock-fields.yml [hunk 2]
```diff
@@ -4,18 +4,16 @@
 - name: forgerock.level
   type: keyword
   description: The log level.
-- name: forgerock.before
+- name: forgerock.before.*
   type: object
+  object_type: keyword
+  object_type_mapping_type: '*'
   description: Specifies the JSON representation of the object prior to the activity.
-- name: forgerock.before.sunAMAuthInvalidAttemptsData
```
Comment (efd6) [inline packages/forgerock/data_stream/am_activity/fields/forgerock-fields.yml:10]: Deleted as it is only here for testing AFAICS. Same for the `... .after. ...` case.

File: packages/forgerock/data_stream/am_activity/fields/forgerock-fields.yml
Comment (efd6) [inline packages/forgerock/data_stream/am_activity/fields/forgerock-fields.yml:1]:
```
      "dynamic_templates": [
        {
          "strings_as_keyword": {
            "match_mapping_type": "string",
            "mapping": {
              "ignore_above": 1024,
              "type": "keyword"
            }
          }
        }
      ],
```

**Review by kgeller — APPROVED** [review]
LGTM

If you're interested in even more validation, I think we have creds in the vault for the testing instance they gave access to for development

**Pull request merged**